### DISCUSSION TITLE
Docker copy fails with permission denied

### DISCUSSION THREAD

**Discussion by prary** [kiali/kiali#4321] [Q&A]
Running ```CLUSTER_TYPE=minikube MINIKUBE_PROFILE=ci MOLECULE_USE_DEV_IMAGES=true MOLECULE_SCENARIO="jaeger-test" make  molecule-test```
fails with following error.
'''
docker cp "/home/arya/.kube/config" molecule-volume-helper:/data/kubeconfig
ERRO[0000] Can't add file /home/arya/.kube/config to tar: open /home/arya/.kube/config: permission denied 
'''

https://github.com/kiali/kiali/blob/2c22b05724f967a95f3204130ce5327c6e9b4a0b/make/Makefile.molecule.mk#L149

Temporary fix for me was to copy 
```
cp .kube/config ~/config
docker cp config molecule-volume-helper:/data/kubeconfig
```

**Comment by jmazzitelli** [comment]
Looks like you just have a permissions restriction on your `/home/arya/.kube/config` file.

**Comment by prary** [comment]
That's what I thought, have tried with chmod+777 but it fails with same error. Not sure why?

**Comment by jmazzitelli** [comment]
https://github.com/docker/for-linux/issues/564

https://stackoverflow.com/questions/45236106/why-does-docker-cp-of-directory-hierarchy-out-of-container-fail-with-permission

Moving this to "Discussions" - doesn't seem to be an issue with anything in Kiali

**Comment by 1057105012** [comment]
sudo addgroup --system docker
sudo adduser $USER docker
newgrp docker
sudo snap disable docker
sudo snap enable docker

you can try it and see docker work normal, i think it's snap design error
